src/rewriter/mod.rs

```diff
@@ -371,9 +380,41 @@ mod tests {
         out
     }
 
+    #[allow(clippy::drop_non_drop)]
+    #[test]
+    fn handlers_covariance() {
```

Comment:
What's the purpose of this test? It doesn't seem to have any assertions

Reply:
It checks the the handlers continue to be covariant, that is, if you have a handler with a lifetime larger than `'a` you can put that handler in a place where a lifetime of `'a` is expected.

The test doesn't assert anything because it is a "if it compile it is good". If the handlers stopped being covariant in their lifetime it test would not compile.

I'll add some comments.